src/bam/mod.rs

```diff
@@ -420,10 +427,10 @@ impl IndexedReader {
         data: *mut ::std::os::raw::c_void,
         record: *mut htslib::bam1_t,
     ) -> i32 {
-        let _self = unsafe { &*(data as *mut Self) };
```

Comment:
This is also weird: these `pileup_read` C functions cast to mutable self, dereference (now we have a copy?) then take the reference of that.

I converted these to `as_ref().unwrap()` or `as_mut().unwrap()` as appropriate.